diff --git a/CMakeExtensions/FindSCIP.cmake b/CMakeExtensions/FindSCIP.cmake
index 6b8b5fc..861bb95 100644
--- a/CMakeExtensions/FindSCIP.cmake
+++ b/CMakeExtensions/FindSCIP.cmake
@@ -5,7 +5,6 @@
 unset(SCIP_LIBRARY CACHE)
 #set(SCIP_INCLUDE_DIRS "" CACHE STRING "the paths to the scip headers")
 unset(SCIP_INCLUDE_DIR CACHE)
-unset(SOPLEX_LIBRARY CACHE)
 unset(Z_LIBRARY CACHE)
 
 if(NOT SCIP_ROOT_DIR)
@@ -25,12 +24,6 @@ find_library(SCIP_LIBRARY scip
 )
 #message(STATUS "SCIP Library: ${SCIP_LIBRARY}")
 
-find_library(SOPLEX_LIBRARY soplex
-    HINTS ${SCIP_ROOT_DIR}/lib64
-          ${SCIP_ROOT_DIR}/lib
-)
-#message(STATUS "Soplex Library: ${SOPLEX_LIBRARY}")
-
 find_library(Z_LIBRARY z
     HINTS ${SCIP_ROOT_DIR}/lib64
           ${SCIP_ROOT_DIR}/lib
@@ -40,7 +33,7 @@ find_library(Z_LIBRARY z
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(SCIP
         DEFAULT_MSG
-        SCIP_LIBRARY SOPLEX_LIBRARY Z_LIBRARY
+        SCIP_LIBRARY Z_LIBRARY
         SCIP_INCLUDE_DIR )
 
 #message(STATUS "SCIP_FOUND: " ${SCIP_FOUND})
@@ -49,12 +42,12 @@ IF (SCIP_FOUND)
     message(STATUS "Found SCIP")
     set(USE_SCIP ON CACHE BOOL "Switching SCIP on")
     SET(SCIP_INCLUDE_DIRS ${SCIP_INCLUDE_DIR})
-    SET(SCIP_LIBRARIES ${SCIP_LIBRARY} ${SCIP90_LIBRARY} ${SOPLEX_LIBRARY} ${Z_LIBRARY})
+    SET(SCIP_LIBRARIES ${SCIP_LIBRARY} ${SCIP90_LIBRARY} ${Z_LIBRARY})
     IF (CMAKE_SYSTEM_NAME STREQUAL "Linux")
     ENDIF (CMAKE_SYSTEM_NAME STREQUAL "Linux")
 ELSE()
     unset(USE_SCIP CACHE)
-    message("SCIP was not found and is not used, if you want to use it, pass SCIP install directory by -DSCIP_ROOT_DIR=... or the individual lib and include directories by -DSCIP_LIBRARIES=... and -DSCIP_INCLUDE_DIRS=...  (Header: ${SCIP_INCLUDE_DIR}, Libraries: ${SCIP_LIBRARY} ${SOPLEX_LIBRARY} ${Z_LIBRARY})")
+    message("SCIP was not found and is not used, if you want to use it, pass SCIP install directory by -DSCIP_ROOT_DIR=... or the individual lib and include directories by -DSCIP_LIBRARIES=... and -DSCIP_INCLUDE_DIRS=...  (Header: ${SCIP_INCLUDE_DIR}, Libraries: ${SCIP_LIBRARY} ${Z_LIBRARY})")
 ENDIF (SCIP_FOUND)
 
-mark_as_advanced(SCIP_INCLUDE_DIR SCIP_LIBRARY SOPLEX_LIBRARY Z_LIBRARY)
+mark_as_advanced(SCIP_INCLUDE_DIR SCIP_LIBRARY Z_LIBRARY)
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1311d42..8df7666 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,10 @@ project (ScaLP)
 
 set(CMAKE_MACOSX_RPATH 1)
 
+if (LINUX)
+    set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON)
+endif()
+
 # Syntax errors are fatal
 cmake_policy(SET CMP0010 NEW)
 
@@ -119,6 +123,10 @@ add_library(ScaLP ${LIBRARY_TYPE}
   ${PARSER_SOURCES}
 )
 
+if (LINUX)
+    set_target_properties(ScaLP PROPERTIES INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
+endif()
+
 #####################
 # Backend-Libraries #
 #####################
