--- /home/pierre/Repositories/flopoco/dev/master/build/dependencies/src/scalp/CMakeExtensions/FindGurobi.cmake	2024-05-30 18:19:04.248830341 +0200
+++ FindGurobi.cmake	2024-05-30 18:18:41.262961201 +0200
@@ -10,6 +10,10 @@
     message(STATUS "Gurobi Root Dir not set by user.")
 else ()
     message(STATUS "Gurobi Root Dir set by user: " ${GUROBI_ROOT_DIR})
+    # Finds out which Gurobi Version is linked:
+    string(REGEX MATCH "[0-9]+" GUROBI_VERSION "${GUROBI_ROOT_DIR}")
+    string(SUBSTRING ${GUROBI_VERSION} 0 2 GUROBI_VERSION_SHORT_1)
+    string(SUBSTRING ${GUROBI_VERSION} 0 3 GUROBI_VERSION_SHORT_2)
 endif()
 
 unset(GUROBI_CPP_INCLUDE_DIR CACHE)
@@ -17,11 +21,6 @@
 unset(GUROBI_CPP_LIBRARY CACHE)
 unset(GUROBI_C_LIBRARY CACHE)
 
-# Finds out which Gurobi Version is linked:
-string(REGEX MATCH "[0-9]+" GUROBI_VERSION "${GUROBI_ROOT_DIR}")
-string(SUBSTRING ${GUROBI_VERSION} 0 2 GUROBI_VERSION_SHORT_1)
-string(SUBSTRING ${GUROBI_VERSION} 0 3 GUROBI_VERSION_SHORT_2)
-
 find_path(GUROBI_CPP_INCLUDE_DIR gurobi_c++.h
         HINTS ${GUROBI_ROOT_DIR}/include
         $ENV{GUROBI_HOME}
